lib/facter/custom_facts/util/fact.rb

```diff
@@ -85,8 +85,9 @@ def define_resolution(resolution_name, options = {}, &block)
 
         resolve
       rescue StandardError => e
-        log.log_exception("Unable to add resolve #{resolution_name.inspect} for fact #{@name}: #{e.message}")
-        nil
+        msg = "Unable to add resolve #{resolution_name.inspect} for fact '#{@name}': #{e.message}"
+        msg += "\n" + e.backtrace.join("\n") if Options[:trace]
+        log.error(msg, true)
```

Comment:
Because of ruby's "return whatever was last evaluted" I think we want to do to preserve compatibility:

```suggestion
        log.error(msg, true)
        nil
```